libs/isograph-react/src/core/areEqualWithDeepComparison.ts

```diff
@@ -61,18 +80,42 @@ export function areEqualObjectsWithDeepComparison(
     return false;
   }
 
-  for (const oldKey of oldKeys) {
-    if (!(oldKey in newItemObject)) {
-      return false;
-    }
-    // @ts-expect-error
-    const oldValue = oldItemObject[oldKey];
-    // @ts-expect-error
-    const newValue = newItemObject[oldKey];
+  for (const field of ast) {
```

Comment:
Oops, they're not unused. But we no longer care to check the `.length` stuff, since we're assuming that `oldItemObject` and `newItemObject` were both created by Isograph. So we can remove the length check too